CONTRIBUTING.md

```diff
@@ -23,66 +23,48 @@ maximize the chances of your PR being merged.
 
 # Breaking change policy
 
-* As of the 1.3.0 release, the Envoy user-facing configuration and APIs are
-  locked and we will not make breaking changes between official numbered
-  releases. This includes bootstrap configuration, REST/gRPC APIs (EDS, CDS, RDS,
-  etc.), and CLI switches. We will also try to not change behavioral semantics
-  (e.g., HTTP header processing order), though this is harder to outright
-  guarantee.
-* We reserve the right to deprecate configuration, after two release cycles. For example, all
-  deprecations between 1.3.0 and 1.4.0 will be deleted soon AFTER 1.5.0 is tagged and released
-  (at the beginning of the 1.6.0 release cycle). This results in a three to six month window for
-  migrating from deprecated code paths to new code paths.
+Both API and implementation stability are important to Envoy. Since the API is consumed by clients
+beyond Envoy, it has a distinct set of [versioning guidelines](api/API_VERSIONING.md). Below, we
+articulate the Envoy implementation stability rules, which operate within the context of the API
+versioning guidelines:
+
+* Features may be marked as deprecated in a given versioned API at any point in time, but this may
+  only be done when a replacement implementation and configuration path is available in Envoy on
+  master. The litmus test is to imagine that a stateless translation tool exists that
+  could convert from the earlier API to the new API. A field may be deprecated
+  if this tool would be able to perform the conversion. For example, removing a
+  field to describe HTTP/2 window settings is valid if a more comprehensive
+  HTTP/2 protocol options field is being introduced to replace it. The PR author
+  deprecating the old configuration is responsible for updating all tests and
```

Comment:
As long as we're in here, I wouldn't mind stronger language here around this.

What we really want is them to update all existing tests to the new configuration, then clone as many tests as needed for coverage and regression testing and guard those. The end state should be that when the deprecated field is removed we're comfortable removing all the DEPRECATED_FEATURE_TEST with no loss of functionaliy.

What we don't want is folks macro guarding most of the tests and us having to do the work later and I don't think I made it as clear as I'd like it on the first pass of this section